homeassistant/util/ssl.py

```diff
@@ -164,3 +151,59 @@ def server_context_intermediate() -> ssl.SSLContext:
     context.set_ciphers(SSL_CIPHER_LISTS[SSLCipherList.INTERMEDIATE])
 
     return context
+
+
+def server_context_insecure() -> ssl.SSLContext:
+    """Return an SSL context for old SSL/TLS standards."""
+    context = ssl.SSLContext(ssl.PROTOCOL_TLS_SERVER)
+
+    context.options |= (
+        ssl.OP_NO_SSLv2 | ssl.OP_NO_SSLv3 | ssl.OP_CIPHER_SERVER_PREFERENCE
+    )
+    if hasattr(ssl, "OP_NO_COMPRESSION"):
+        context.options |= ssl.OP_NO_COMPRESSION
+
+    context.set_ciphers(SSL_CIPHER_LISTS[SSLCipherList.INSECURE])
+
+    return context
+
+
+# Create this only once and reuse it
+_DEFAULT_SSL_CONTEXT = client_context()
+_DEFAULT_NO_VERIFY_SSL_CONTEXT = create_no_verify_ssl_context()
+_NO_VERIFY_SSL_CONTEXTS = {
+    SSLCipherList.INTERMEDIATE: create_no_verify_ssl_context(
+        SSLCipherList.INTERMEDIATE
+    ),
+    SSLCipherList.MODERN: create_no_verify_ssl_context(SSLCipherList.MODERN),
+    SSLCipherList.INSECURE: create_no_verify_ssl_context(SSLCipherList.INSECURE),
+}
+_SSL_CONTEXTS = {
+    SSLCipherList.INTERMEDIATE: server_context_intermediate(),
+    SSLCipherList.MODERN: server_context_modern(),
+    SSLCipherList.INSECURE: server_context_insecure(),
```

Comment:
It looks like we are returning Server contexts for the aiohttp client?

```
Deprecated since version 3.10: [SSLContext](https://docs.python.org/3/library/ssl.html#ssl.SSLContext) without protocol argument is deprecated. The context class will either require [PROTOCOL_TLS_CLIENT](https://docs.python.org/3/library/ssl.html#ssl.PROTOCOL_TLS_CLIENT) or [PROTOCOL_TLS_SERVER](https://docs.python.org/3/library/ssl.html#ssl.PROTOCOL_TLS_SERVER) protocol in the future.
```

Reply:
Ah right, now I get the diffrence, I was already confused.
I changed it now to return the client_contexts.

It looks better now.

@bdraco do you think we can get this merged before the beta cut?
Otherwise I will need to revert a Reolink PR because it is now broken and to fix it properly I need these SSL_CONTEXTS.